test/runtests.jl

```diff
@@ -89,3 +92,77 @@ end
     @test in(TimeSpan(1,2))(Nanosecond(1))
     @test !in(TimeSpan(1,2))(Nanosecond(2))
 end
+
+@testset "Set operations: (e.g. `intersect`, `union`, `setdiff`)" begin
+    
+    myduration(x::TimeSpan) = duration(x)
+    myduration(x::AbstractVector{TimeSpan}) = reduce(+, map(duration, x), init = Nanosecond(0))
+    myunion(x::TimeSpan) = x
+    myunion(x::AbstractVector{TimeSpan}) = reduce(∪, x)
+    function testsets(a, b)
+        @test myduration((a ∪ b)) ≤ myduration(myunion(a)) + myduration(myunion(b))
+        @test myduration(setdiff(a, b)) ≤ myduration(myunion(a))
+        @test myduration((a ∩ b)) + myduration(symdiff(a, b)) ==
+            myduration(union(a,b))
+        @test a ⊆ (a ∪ b)
+        @test !issetequal(a, setdiff(a, b))
+        @test issetequal(a, a)
+        @static if VERSION ≥ v"1.5"
+            @test isdisjoint(setdiff(a, b), b)
+            @test !isdisjoint(a, a)
+        end
+    end
+
+    @test isempty(reduce(∪, TimeSpan[]))
+
+    spans = [TimeSpan(Nanosecond(i), Nanosecond(i+3)) for i in 1:5:15]
+    spans = [spans; translate.(spans, Nanosecond.(1:2:5))]
+    @test all(start.(spans) .∈ Ref(spans))
+    testsets(spans[1:3], spans[4:end])
+    testsets(spans[1], spans[4:end])
+    testsets(spans[1:3], spans[4])
+
+    # verify that `stop` need not be ordered
+    spans = [TimeSpan(Nanosecond(0), Nanosecond(5)), TimeSpan(Nanosecond(0), Nanosecond(3))]
+    @test shortest_timespan_containing(union(spans)) == TimeSpan(Nanosecond(0), Nanosecond(5))
+
+    starts = Nanosecond.(rand(1:100_000, 25))
+    spans = TimeSpan.(starts, starts .+ Nanosecond.(rand(1:10_000)))
```

Comment:
Lets avoid being misleading:
```suggestion
    spans = TimeSpan.(starts, starts .+ Nanosecond(rand(1:10_000)))
```